File: BankAccount.java
/*
Write a Menu driven program in java to implement simple banking application.
Application should read the customer name, account number, initial balance, rate of interest, contact number and address field etc.
Application should have following methods.
1. createAccount()
2. deposit()
3. withdraw()
4. computeInterest()
5. displayBalance()
*/

//package dbit;
import java.util.Scanner;
public class BankAccount 
{
	double initialbalance,finalbalance,amount;
	int accountno,contact;
	float rate;
	String name,address;
	static Scanner sc=new Scanner(System.in);
	
	public static void main(String args[])
	{
		
        int p;
	    int ch;
	    BankAccount a=new BankAccount();
        do
        {
           
			System.out.println("   MENU    ");
			System.out.println("1. Create an Account");
			System.out.println("2. Deposit Money");
			System.out.println("3. Withraw Money");
			System.out.println("4. Compute Interest");
			System.out.println("5. Display the Total Amount");
			ch=sc.nextInt();
			switch(ch)
			{
                 case 1:a.createAccount();
						System.out.println("Your account details:");
						System.out.println("Name:"+a.name);
                        System.out.println("Contact no.:"+a.contact);
                        System.out.println("Address:"+a.address);
                        System.out.println("Account No. :"+a.accountno);
                        break;
               	 case 2:double t=a.deposit();
			   			System.out.println("Your total amount in the account:"+t);
                        break;
               	 case 3:double d=a.withdraw();
			  			System.out.println("Your total amount in the account:"+d);
                        System.out.println("Rs." + a.finalbalance + " has been withdrawn.");
			    		break;
				 case 4:double c=a.computeInterest();
						System.out.println("Your total amount in the account:"+c);
						break;
			     case 5:a.displayBalance();
					    break;
			}
			System.out.println("Do you wish to continue?"+"\n"+"Press 1 for Yes & 0 for No");
			p=sc.nextInt();
		}	
		while(p!=0);
	}
	void createAccount()
	{
	    System.out.println("Enter your name :");
	    name=sc.next();
	    System.out.println("Enter your contact no.:");
		contact=sc.nextInt();
		System.out.println("Enter your address field :");
		address=sc.next();
		System.out.println("Enter your account no. :");
		accountno=sc.nextInt();
		System.out.println("Enter your initial balance:");
		initialbalance=sc.nextDouble();
	}
	double deposit()
	{
		System.out.println("Enter your Amount you've deposited:");
		amount=sc.nextDouble();
		finalbalance=initialbalance + amount;
		initialbalance=finalbalance;
		return finalbalance;
	}
	double withdraw()
	{
		System.out.println("Enter your Amount you've withdrawn:");
		amount=sc.nextDouble();
		finalbalance=initialbalance - amount;
		initialbalance=finalbalance;
        return finalbalance;
	}
	double computeInterest()
	{
		double interest;
		System.out.println("Enter your interest:");
		interest=sc.nextDouble();
		finalbalance=initialbalance + interest*initialbalance;
		initialbalance=finalbalance;
        return finalbalance;
	}
	void displayBalance()
	{
                System.out.println("Your total amount in the account:"+finalbalance);
	}
}

File: bankAccount.py
class BankAccount:
    def __init__(self):
        self.initialbalance = 0.0
        self.finalbalance = 0.0
        self.amount = 0.0
        self.accountno = 0
        self.contact = 0
        self.rate = 0.0
        self.name = ""
        self.address = ""
    
    def create_account(self):
        self.name = input("Enter your name: ")
        self.contact = int(input("Enter your contact no.: "))
        self.address = input("Enter your address: ")
        self.accountno = int(input("Enter your account no.: "))
        self.initialbalance = float(input("Enter your initial balance: "))
    
    def deposit(self):
        self.amount = float(input("Enter the amount you've deposited: "))
        self.finalbalance = self.initialbalance + self.amount
        self.initialbalance = self.finalbalance
        return self.finalbalance
    
    def withdraw(self):
        self.amount = float(input("Enter the amount you've withdrawn: "))
        self.finalbalance = self.initialbalance - self.amount
        self.initialbalance = self.finalbalance
        return self.finalbalance
    
    def compute_interest(self):
        interest = float(input("Enter your interest rate: "))
        self.finalbalance = self.initialbalance + interest * self.initialbalance
        self.initialbalance = self.finalbalance
        return self.finalbalance
    
    def display_balance(self):
        print("Your total amount in the account: ", self.finalbalance)

def main():
    a = BankAccount()
    while True:
        print("   MENU    ")
        print("1. Create an Account")
        print("2. Deposit Money")
        print("3. Withdraw Money")
        print("4. Compute Interest")
        print("5. Display the Total Amount")
        ch = int(input("Enter your choice: "))
        
        if ch == 1:
            a.create_account()
            print("Your account details:")
            print(f"Name: {a.name}")
            print(f"Contact no.: {a.contact}")
            print(f"Address: {a.address}")
            print(f"Account No.: {a.accountno}")
        elif ch == 2:
            total = a.deposit()
            print(f"Your total amount in the account: {total}")
        elif ch == 3:
            total = a.withdraw()
            print(f"Your total amount in the account: {total}")
            print(f"Rs. {a.finalbalance} has been withdrawn.")
        elif ch == 4:
            total = a.compute_interest()
            print(f"Your total amount in the account: {total}")
        elif ch == 5:
            a.display_balance()
        else:
            print("Invalid option, please choose again.")
        
        p = int(input("Do you wish to continue?\nPress 1 for Yes & 0 for No: "))
        if p == 0:
            break

if __name__ == "__main__":
    main()
